--- a/vboxdrv/r0drv/linux/time-r0drv-linux.c	2018-12-18 17:53:53.638539634 +0100
+++ b/vboxdrv/r0drv/linux/time-r0drv-linux.c	2018-12-18 18:04:10.629613802 +0100
@@ -171,11 +171,20 @@
 {
     IPRT_LINUX_SAVE_EFL_AC();
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 16)
+/* On Linux 4.20, time.h includes time64.h and we have to use 64-bit times. */
+# ifdef _LINUX_TIME64_H
+    struct timespec64 Ts;
+    ktime_get_real_ts64(&Ts);
+# else
     struct timespec Ts;
     ktime_get_real_ts(&Ts);
+# endif
     IPRT_LINUX_RESTORE_EFL_AC();
+# ifdef _LINUX_TIME64_H
+    return RTTimeSpecSetTimespec64(pTime, &Ts);
+#else
     return RTTimeSpecSetTimespec(pTime, &Ts);
-
+#endif
 #else   /* < 2.6.16 */
     struct timeval Tv;
     do_gettimeofday(&Tv);
