--- a/dkms/vboxguest/5.1.14_OSE/source/vboxsf/lnkops.c	2017-02-11 23:08:57.188914914 +0100
+++ b/dkms/vboxguest/5.1.14_OSE/source/vboxsf/lnkops.c	2017-02-12 09:28:06.848072431 +0100
@@ -90,7 +90,9 @@
 
 struct inode_operations sf_lnk_iops =
 {
+# if LINUX_VERSION_CODE < KERNEL_VERSION(4, 10, 0)
     .readlink       = generic_readlink,
+# endif
 # if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 5, 0)
     .get_link       = sf_get_link
 # elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 2, 0)
