--- a/5.0.0/source/vboxsf/lnkops.c	2015-07-28 10:55:35.446264623 +0200
+++ a/5.0.0/source/vboxsf/lnkops.c	2015-07-28 13:38:45.201931456 +0200
@@ -18,7 +18,7 @@
 
 #include "vfsmod.h"
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 0) && LINUX_VERSION_CODE <= KERNEL_VERSION(4, 1, 99)
 
 static void *sf_follow_link(struct dentry *dentry, struct nameidata *nd)
 {
